### ISSUE TITLE
What is the cleanest way to have multiple Auths in same client ?

### ISSUE THREAD

**Issue by bogas04** [DaftMonk/generator-angular-fullstack#858]
I have different set of users (not roles) and for each category I want a different Auth service. For example the application has agents, companies and admins, so it would have AuthAdmin, AuthCompany and AuthAgent.

If they were on different clients then it wouldn't have been an issue. But I have to use a single client.
Currently I've simply created a different Auth service at both ends, but the issue is that as soon as the application loads (say I open the url in other tab/ have a target="_blank" link), the app calls server to check for authentication, and if the first Auth service fails, it resets all the cookies and tokens, which doesn't let second one log in (considering it was the one having the tokens). How can I fix this ?

tl;dr : Where is the code segment which calls server on application load to check for authentication?
How can I tweak it in a way to check for each Auth service and destroy the cookie tokens ONLY if all of them fail.

Sorry for being less descriptive or being vague in my problem statement. 

What I've done : 
client/app/components/ has auth, auth-company, auth-admin etc, which have the corresponding login.
and similarly different pages are there for their login. 

[Stop api/admin/me from being called](https://cloud.githubusercontent.com/assets/6177621/6430268/801a580c-c028-11e4-8637-4a29c6c30ccd.png)

**Comment by bogas04** [comment]
Here's my app.js https://gist.github.com/bogas04/ec8bb031e169a4a76e5d

**Comment by remicastaing** [comment]
You could use Q.any() of [Q promise](https://github.com/kriskowal/q) on the client side to check all Auth services.

The any function accepts an array of promises and returns a promise that is fulfilled by the first given promise to be fulfilled, or rejected if all of the given promises are rejected.

**Comment by DaftMonk** [comment]
The cookie is removed in the app.js response error interceptor. Normally it only checks if there was a 401 before deleting the cookie and redirecting, but in your case you could pass the response into a service that checks if the user really does need to be re-authenticated before removing the cookie and redirecting them.

In the new service whenever a failure comes in, you could update an object that keeps track of the state of failed requests from specific urls, and if all three auth urls that you care about fail, you then delete the cookie and send the user to the appropriate login page
